utils/dataHandling.ts

```diff
@@ -0,0 +1,277 @@
+            unsub?.(); 
+            clearInterval(timer);
+            resolve(value);
+          }
+        }
+      });
+
+      timer = setInterval(() => {
+        if (Date.now() - start > timeoutMs) {
+          if (!done) {
+            done = true;
+            console.warn('[PROFILE] Timeout waiting for profile');
+            unsub?.();
+            clearInterval(timer);
+            reject(new Error('Timed out waiting for profile'));
+          }
+        }
+      }, pollInterval);
+    });
+
+    return profile;
+  } catch (error: any) {
+    console.error('[PROFILE] Error loading profile:', error);
+    throw new Error(error.message || 'Failed to load profile');
+  }
+}; 
+
+/**
+ * Updates the user's profile name by publishing a kind 0 metadata event
+ * @param newName - The new name to set
+ * @param currentProfile - Optional current profile data to avoid reloading
+ * @returns Promise that resolves when the event is published
+ */
+export const changeName = async (newName: string, currentProfile?: any): Promise<void> => {
+  if (!newName || newName.trim() === '') {
+    throw new Error('Name cannot be empty');
+  }
+  try {
+    console.log('[PROFILE] Updating name to:', newName);
+    
+    const currentPubkey = pubkey.get();
+    if (!currentPubkey) {
+      throw new Error('No user logged in');
+    }
+
+    //get total profile data so updating doesnt lose anything
+    let profileData = currentProfile;
+    if (!profileData) {
+      profileData = await loadUserProfile(currentPubkey);
+    }
+    console.log('[PROFILE] Current profile data:', profileData);
+
+    const profileContent = JSON.stringify({
+      ...profileData, // Preserve existing fields
+      name: newName.trim(), // Update only the name
+    });
+    
+    const profileEvent = makeEvent(0, { content: profileContent });
+
+    // Get relays to publish to
+    const router = Router.get();
+    const publishRelays = router.FromUser().getUrls();
+    
+    if (publishRelays.length === 0) {
+      console.warn('[PROFILE] No publish relays found, using default relays');
+      publishRelays.push("wss://relay.damus.io/", "wss://nos.lol/");
+    }
+
+    console.log('[PROFILE] Publishing to relays:', publishRelays);
+
+    publishThunk({
+      relays: publishRelays,
+      event: profileEvent,
+      delay: 3000, 
+    });
+
+    console.log('[PROFILE] Name update published successfully');
+  } catch (error: any) {
+    console.error('[PROFILE] Error updating name:', error);
+    throw new Error(error.message || 'Failed to update name');
+  }
+}; 
+
+/**
+ * Updates the user's profile description by publishing a kind 0 metadata event
+ * @param newDescription - The new description/about text to set
+ * @param currentProfile - Optional current profile data to avoid reloading
+ * @returns Promise that resolves when the event is published
+ */
+export const changeDescription = async (newDescription: string, currentProfile?: any): Promise<void> => {
```

Comment:
Because this is so similar to changeName, and because `Profile` is a known type, it's best to just do a single function to publish the event. I've added this to welshman as `setProfile` (although we'll eventually have to add a custom version like the one in flotilla).